packages/jsii-diff/test/classes.test.ts

```diff
@@ -359,20 +359,83 @@ describe('implement base types need to be present in updated type system', () =>
 
 // ----------------------------------------------------------------------
 
-test('cannot make a class property optional', () =>
+test.each([
+  [
+    'name: string',
+    'name?: string',
+    /type Optional<string> \(formerly string\): output type is now optional/,
+  ],
+  [
+    'name?: string',
+    'name: string',
+    undefined, // Strengthening is okay
+  ],
+  [
+    'name: string',
+    'name: string | number',
+    /string \| number is not assignable to string/,
+  ],
+  [
+    'name: string | number',
+    'name: string',
+    undefined, // Strengthening is okay
+  ],
+])('change class property ', (oldDecl, newDecl, error) =>
```

Comment:
I kinda would have preferred using objects rather than tuples for this (since this is string-abundant).

```suggestion
test.each([
  {
    oldDecl: 'name: string',
    newDecl: 'name?: string',
    error: /type Optional<string> \(formerly string\): output type is now optional/,
  },
  {
    oldDecl: 'name?: string',
    newDecl: 'name: string',
    error: undefined, // Strengthening is okay
  },
  {
    oldDecl: 'name: string',
    newDecl: 'name: string | number',
    error: /string \| number is not assignable to string/,
  },
  {
    oldDecl: 'name: string | number',
    newDecl: 'name: string',
    error: undefined, // Strengthening is okay
  },
])('change class property ', ({oldDecl, newDecl, error}) =>
```

The same applies to subsequent similar blocks.

Reply:
Cool, never even considered to do this.